src/catalog/src/memory/objects.rs

```diff
@@ -693,27 +693,29 @@ impl mz_sql::catalog::CatalogCollectionItem for CatalogCollectionEntry {
             CatalogItem::Source(source) => source.global_id,
             CatalogItem::Log(log) => log.global_id,
             CatalogItem::View(view) => view.global_id,
-            CatalogItem::MaterializedView(mv) => mv.global_id,
             CatalogItem::Sink(sink) => sink.global_id,
             CatalogItem::Index(index) => index.global_id,
             CatalogItem::Type(ty) => ty.global_id,
             CatalogItem::Func(func) => func.global_id,
             CatalogItem::Secret(secret) => secret.global_id,
             CatalogItem::Connection(conn) => conn.global_id,
             CatalogItem::ContinualTask(ct) => ct.global_id,
-            CatalogItem::Table(table) => match self.version {
+            CatalogItem::MaterializedView(_) | CatalogItem::Table(_) => match self.version {
                 RelationVersionSelector::Latest => {
-                    let (_version, gid) = table
-                        .collections
+                    let (_version, gid) = self
+                        .item
+                        .versions()
+                        .expect("materialized views and tables have versions")
                         .last_key_value()
                         .expect("at least one version");
```

Comment:
This could just be `self.item.latest_global_id()`, right?

Reply:
Yep, and cleaned up related functions.